packages/extension/src/newtab/ShortcutLinks/experiments/ShortcutLinksUIV1.tsx

```diff
@@ -121,12 +122,18 @@ export function ShortcutLinksUIV1(props: ShortcutLinksV1Props): ReactElement {
     shouldUseListFeedLayout,
     showTopSites,
     toggleShowTopSites,
+    hasCheckedPermission,
   } = props;
 
   const { checkHasCompleted, isActionsFetched, completeAction } = useActions();
   const { githubShortcut } = useThemedAsset();
 
+  if (!hasCheckedPermission) {
+    return null;
+  }
```

Comment:
This is basically loading state for top sites so I just don't render anything until browser check has passed to avoid flash of popup.